dev/io.openliberty.checkpoint_fat/test-applications/ejbapp1/src/ejbapp1/TimerTest.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2015, 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package ejbapp1;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+import javax.ejb.Schedule;
+import javax.ejb.Singleton;
+
+@Singleton
+public class TimerTest {
+    private final AtomicLong lastRun = new AtomicLong(-1);
+    private final AtomicInteger numRun = new AtomicInteger(0);
+    private final AtomicBoolean failed = new AtomicBoolean(false);
+
+    @Schedule(hour = "*", minute = "*", second = "*/1", persistent = false)
+    public void everyOneSeconds() {
+        long currentTime = System.nanoTime();
+        long lastTime = lastRun.getAndSet(currentTime);
+        if (lastTime == -1) {
+            return;
+        }
+        long deltaTime = currentTime - lastTime;
+        long deltaMillis = TimeUnit.NANOSECONDS.toMillis(deltaTime);
+        // here we assume if delta is less than 700 millis then something is wrong
+        if (deltaMillis < 800) {
+            failed.set(true);
+        }
+        if (numRun.addAndGet(1) == 5) {
+            System.out.println("TIMER RUN 5 TIMES");
```

Comment:
I would recommend that this method be changed to accept a Timer parameter, and then add a timer.cancel() here, once the timer has expired 5 times.  While the test should work as coded, there will likely be intermittent build break issues opened due to the timer continuing to run when the test finishes and attempts to shutdown the server.

Under normal conditions, the test will probably complete and shutdown the server before the timer runs again; however, due to the sometimes erratic performance behavior of the test systems, there can be pauses that will result in the timer running before shutdwon, or more specifically during the server shutdown.  While the EJB Timer Service blocks timers running once server shutdown is initiated, if a timer is already running, it can result in various errors and or FFDC, which would need to be allowed by the test.  Adding a timer.cancel() eliminates that concern.